packages/auth/src/access/service.ts

```diff
@@ -63,3 +65,14 @@ async function getByGrant(
     grantId
   })
 }
+
+function validateLimits(accessRequests: AccessRequest[]) {
+  const areBothLimitsSet =
+    accessRequests.findIndex(
+      (access) => access.limits?.debitAmount && access.limits.receiveAmount
+    ) !== -1
+
+  if (areBothLimitsSet) {
+    throw new Error('Only one of receiveAmount or debitAmount allowed')
+  }
+}
```

Comment:
Following the stack trace for the error, we would end up sending a 500 - Internal server error back to the client.
Instead, we should reply with a 400  `invalid_request`.